src/node/iir_filter.rs

```diff
@@ -278,268 +306,398 @@ impl AudioProcessor for IirFilterRenderer {
         let input = &inputs[0];
         let output = &mut outputs[0];
 
-        self.filter(input, output);
+        // handle tail time
+        if input.is_silent() {
+            let mut ended = true;
+
+            // if all values in states are 0., we have nothing left to process
+            self.states.iter().all(|state| {
+                if state.iter().any(|&v| v.is_normal()) {
+                    ended = false;
+                }
+                // if ended is false, `iter().all` will stop early
+                ended
+            });
+
+            if ended {
+                output.make_silent();
+                return false;
+            }
+        }
 
-        true // todo tail time - issue #34
-    }
-}
+        // eventually resize state according to input number of channels
+        // if in tail time we should continue with previous number of channels
+        if !input.is_silent() {
```

Comment:
Nitpick, maybe swap the if/else as
```
if input.is_silent {
 // silent case
} else {
 // non-silent
}
```

Parsing the else was a bit hard to me, double negation